### PR TITLE
Dynamic userid handling

### LINKED ISSUE #45: one Home Assistant, multiple grocy Users?
Hi everybody,

I just installed the grocy addon and this chores card. It works great, but there is one thing I couldn't figure out (or is it not implemented?): multiple people use Home Assistant, but when either of those click `track` on the card, it always gets marked as done by me.

Each person has their own Home Assistant account, but the integration itself has been created by me (and is using that API key). Is there a way to pass on which user has actually completed a chore via this card, so that grocy will be able to mark the correct person?

Thank you in advance for your help :)

### LINKED ISSUE #95: Mark chores as completed by current HA user
I have implemented the Grocy Chores card using an API key for a generic user in Grocy. Whenever I complete a chore or task from HA it is recorded in Grocy as completed by that generic user.
I would like to submit a wish for the completion to be recorded as performed by the current user in HA, assuming that the username is the same between HA and Grocy.

### PR DESCRIPTION
**Pull request by karwosts**
Allow using the current hass user for chore/task tracking, filtering. and creating new tasks.

To setup, in the config for user_id, setup a map from homeassistant usernames to grocy user_ids:

```
user_id:
  tim: 1
  bob: 2
  alice: 3
  default: 0
 ```
  
 Chores/Tasks will be tracked by the user_id that matches the username of the current logged in user. 
 
 If `filter_user` is set to the value "current", it will filter to the tasks/chores assigned to the current hass user. 

Still need to update the README for this behavior. I only really have one hass/grocy user, so this could probably use some extra testing, but it looked good in the debugger. 

Fix #45
Fix #95

Basically just eliminate the need for state switch.

### CODE REVIEW

**Comment by isabellaalstrom** [comment]
Awesome work!

**Comment by larsalthof** [comment]
This is absolutely awesome. Using the currently logged in user to complete chores works, but I am not sure it works for the filter=current for tasks.
I have a card from earlier:
image
and a new card:
image

I would expect them to give the same result for myself, but...
image

**Comment by karwosts** [comment]
Ah I actually didn't realize that, I guess in my testcase the username and name were always the same. Good to know.

**Comment by DeastinY** [comment]
I can not get this to work. I have updated to 3.8.0, tried both capitalized usernames and lower case ones. Filtering with current_user always fails and the tasks get tracked with the same (first) user_id always.

```
type: custom:grocy-chores-card
title: Chores
entity:
  - sensor.grocy_chores
show_quantity: 5
show_days: 5
show_last_tracked: false
show_chores_without_due: false
show_divider: true
due_in_days_threshold: 10
filter_user: current_user <--- not showing any tasks then
user_id:
  jim: 1    <--- everything tracked as jim
  jack: 2
  john: 3
  joe: 4
```

**Comment by karwosts** [comment]
> I can not get this to work.

Can you identify a specific chore that's not working as expected, and then can you copy and show the attributes for a specific chore from `sensor.grocy_chores` that's not working, and include a screenshot of your users table from ha settings?

**Comment by larsalthof** [comment]
> I can not get this to work. I have updated to 3.8.0, tried both capitalized usernames and lower case ones. Filtering with current_user always fails and the tasks get tracked with the same (first) user_id always.

Are your usernames and names the same? This actually uses the name, not the username.
